app/models/purchase.py

```diff
@@ -20,10 +21,9 @@ class CartItem(db.Model):
     listing = db.relationship("Listing")
 
     @staticmethod
-    def delete_cart_items():
-        for cart_item in current_user.cart_items:
+    def delete_cart_items(cart_items):
+        for cart_item in cart_items:
             db.session.delete(cart_item)
-        db.session.commit()
```

Comment:
We want to commit deleting the cart items along with the other database changes we make (creating the purchases & the order). So these changes ultimately get committed in the commit at the end of `order_cart_items_from_vendor`.